File: backend/app/models/enums.py
from enum import Enum as PyEnum

class BillType(str, PyEnum):
    EXPENSE = "expense"  # 支出
    INCOME = "income"    # 收入

class UserRole(str, PyEnum):
    ADMIN = "admin"      # 管理员
    MEMBER = "member"    # 普通成员

class InvitationStatus(str, PyEnum):
    PENDING = "pending"      # 待接受
    ACCEPTED = "accepted"    # 已接受
    REJECTED = "rejected"    # 已拒绝
    EXPIRED = "expired"      # 已过期

class LedgerStatus(str, PyEnum):
    ACTIVE = "active"        # 活跃
    DELETED = "deleted"      # 已删除（回收站）

class BudgetPeriodType(str, PyEnum):
    MONTHLY = "monthly"      # 月度预算
    QUARTERLY = "quarterly"  # 季度预算
    YEARLY = "yearly"        # 年度预算
    CUSTOM = "custom"        # 自定义周期

class BudgetStatus(str, PyEnum):
    ACTIVE = "active"        # 活跃
    PAUSED = "paused"        # 暂停
    COMPLETED = "completed"  # 已完成
    EXPIRED = "expired"      # 已过期

class AlertType(str, PyEnum):
    WARNING = "warning"      # 预警（80%）
    CRITICAL = "critical"    # 严重（95%）
    EXCEEDED = "exceeded"    # 超支 

File: frontend/src/constants/enums.ts
// 用户角色枚举
export enum UserRole {
  ADMIN = 'admin',
  MEMBER = 'member'
}

// 账单类型枚举
export enum BillType {
  EXPENSE = 'expense',
  INCOME = 'income'
}

// 邀请状态枚举
export enum InvitationStatus {
  PENDING = 'pending',
  ACCEPTED = 'accepted',
  REJECTED = 'rejected',
  EXPIRED = 'expired'
}

// 账本状态枚举
export enum LedgerStatus {
  ACTIVE = 'active',
  DELETED = 'deleted'
}

// 用户状态枚举
export enum UserStatus {
  ACTIVE = 'active',
  INACTIVE = 'inactive'
}

// 预算周期类型枚举
export enum BudgetPeriodType {
  MONTHLY = 'monthly',
  WEEKLY = 'weekly',
  QUARTERLY = 'quarterly',
  YEARLY = 'yearly',
  CUSTOM = 'custom'
}

// 预算状态枚举
export enum BudgetStatus {
  ACTIVE = 'active',
  PAUSED = 'paused',
  COMPLETED = 'completed',
  EXPIRED = 'expired'
}

// 预算提醒类型枚举
export enum AlertType {
  WARNING = 'warning',
  CRITICAL = 'critical',
  EXCEEDED = 'exceeded'
}

// 角色显示信息
export const ROLE_DISPLAY = {
  [UserRole.ADMIN]: {
    text: '管理员',
    description: '可以编辑账本信息、邀请/移除成员、管理账单记录'
  },
  [UserRole.MEMBER]: {
    text: '成员',
    description: '可以查看账本信息、记录和编辑自己的账单'
  }
} as const;

// 邀请状态显示信息
export const INVITATION_STATUS_DISPLAY = {
  [InvitationStatus.PENDING]: {
    text: '待处理',
    color: 'text-yellow-600'
  },
  [InvitationStatus.ACCEPTED]: {
    text: '已接受',
    color: 'text-green-600'
  },
  [InvitationStatus.REJECTED]: {
    text: '已拒绝',
    color: 'text-red-600'
  },
  [InvitationStatus.EXPIRED]: {
    text: '已过期',
    color: 'text-gray-600'
  }
} as const;

// 账本状态显示信息
export const LEDGER_STATUS_DISPLAY = {
  [LedgerStatus.ACTIVE]: {
    text: '活跃'
  },
  [LedgerStatus.DELETED]: {
    text: '已删除'
  }
} as const; 